File: tribonacci/Tribonacci.java
import java.util.Arrays;
import java.util.stream.IntStream;

public class Tribonacci {
    public static void main(String[] args) {
        int[] result = Tribonacci.tribonacci(new int[]{1, 1, 2}, 10);
        System.out.println(Arrays.toString(result));
    }

    public static int[] tribonacci(int[] initialSequence, int newSequenceCount) {
        int sequenceLength = initialSequence.length;
        int[] newSequence = Arrays.copyOf(initialSequence, newSequenceCount);

        for (int index = sequenceLength; index < newSequenceCount; index++) {
            newSequence[index]  = IntStream
                .of(newSequence)
                .skip(index - sequenceLength)
                .sum();
        }

        return newSequence;
    }
}

File: tribonacci/tribonacci.js
function tribonacci(initialSequence, newSequenceCount) {
  const sequenceLength = initialSequence.length;
  const newSequence = [].concat(initialSequence);

  for (let index = sequenceLength; index < newSequenceCount; index++) {
    const nextNumber = newSequence
      .slice(index - sequenceLength, index)
      .reduce((accumulator, valueToSum) => accumulator + valueToSum, 0);
    newSequence.push(nextNumber);
  }

  return newSequence;
}

console.log(tribonacci([0,1,1], 5));